Packer crashes with port must be specified error when communicator is set to none

Packer version: 1.0.3
Host platform: RHEL7 (on AWS)

The similar issue was referenced in the past: #3054 #2736 #2848 #3304. I am submitting this new issue as was suggested by @rickard-von-essen on #3054 

I can't access gist from my work. Hence pasting the Json, command and debug output here.

elasticbeanstalk.json:
```
{
  "variables": {
     "ami_name_prefix": "",
     "source_ami_prefix": "",
     "region": "",
     "kms_key_id": ""
  },
  "builders": [
    {
      "type": "amazon-ebs",
      "communicator": "none",
      "ami_name": "{{user `ami_name_prefix`}} {{isotime \"2006-01-02 15-04-05\"}}",
      "ami_description": "Created by Packer using amazon-ebs builder",
      "instance_type": "m3.medium",
      "region": "{{user `region`}}",     
      "source_ami_filter": {
        "filters": {
          "virtualization-type": "hvm",
          "name": "{{user `source_ami_prefix`}}",
          "root-device-type": "ebs"
        },
        "owners": ["amazon"],
        "most_recent": true
      },
      "user_data_file": "elasticbeanstalk_amzn_linux.userdata",
      "encrypt_boot": "true",
      "kms_key_id": "{{user `kms_key_id`}}"     
    }
  ]
}
```
  
elasticbeanstalk_amzn_linux.userdata:
#cloud-config
repo_releasever: 2017.03
repo_upgrade: none

Command:
PACKER_LOG=1 packer build \
  -var 'region=us-east-1' \
  -var 'ami_name_prefix=myinc-ElasticBeanstalk-CARS-Encr' \
  -var 'source_ami_prefix=aws-elasticbeanstalk-amzn-*.x86_64-tomcat8java8-hvm-*' \
  -var 'kms_key_id=fffff-aaaa-bbbb-cccc-ddddddd' \
  elasticbeanstalk.json

Debug log:
```
2017/08/03 08:22:52 [INFO] Packer version: 1.0.3
2017/08/03 08:22:52 Packer Target OS/Arch: linux amd64
2017/08/03 08:22:52 Built with Go Version: go1.8.1
2017/08/03 08:22:52 Detected home directory from env var: /home/userxyz
2017/08/03 08:22:52 Using internal plugin for parallels-iso
2017/08/03 08:22:52 Using internal plugin for profitbricks
2017/08/03 08:22:52 Using internal plugin for virtualbox-ovf
2017/08/03 08:22:52 Using internal plugin for vmware-vmx
2017/08/03 08:22:52 Using internal plugin for amazon-chroot
2017/08/03 08:22:52 Using internal plugin for docker
2017/08/03 08:22:52 Using internal plugin for googlecompute
2017/08/03 08:22:52 Using internal plugin for hyperv-iso
2017/08/03 08:22:52 Using internal plugin for null
2017/08/03 08:22:52 Using internal plugin for azure-arm
2017/08/03 08:22:52 Using internal plugin for cloudstack
2017/08/03 08:22:52 Using internal plugin for oneandone
2017/08/03 08:22:52 Using internal plugin for openstack
2017/08/03 08:22:52 Using internal plugin for parallels-pvm
2017/08/03 08:22:52 Using internal plugin for amazon-ebs
2017/08/03 08:22:52 Using internal plugin for amazon-ebsvolume
2017/08/03 08:22:52 Using internal plugin for amazon-instance
2017/08/03 08:22:52 Using internal plugin for qemu
2017/08/03 08:22:52 Using internal plugin for vmware-iso
2017/08/03 08:22:52 Using internal plugin for file
2017/08/03 08:22:52 Using internal plugin for triton
2017/08/03 08:22:52 Using internal plugin for virtualbox-iso
2017/08/03 08:22:52 Using internal plugin for alicloud-ecs
2017/08/03 08:22:52 Using internal plugin for amazon-ebssurrogate
2017/08/03 08:22:52 Using internal plugin for digitalocean
2017/08/03 08:22:52 Using internal plugin for chef-solo
2017/08/03 08:22:52 Using internal plugin for puppet-server
2017/08/03 08:22:52 Using internal plugin for salt-masterless
2017/08/03 08:22:52 Using internal plugin for shell
2017/08/03 08:22:52 Using internal plugin for ansible
2017/08/03 08:22:52 Using internal plugin for ansible-local
2017/08/03 08:22:52 Using internal plugin for chef-client
2017/08/03 08:22:52 Using internal plugin for converge
2017/08/03 08:22:52 Using internal plugin for puppet-masterless
2017/08/03 08:22:52 Using internal plugin for windows-restart
2017/08/03 08:22:52 Using internal plugin for file
2017/08/03 08:22:52 Using internal plugin for powershell
2017/08/03 08:22:52 Using internal plugin for shell-local
2017/08/03 08:22:52 Using internal plugin for windows-shell
2017/08/03 08:22:52 Using internal plugin for amazon-import
2017/08/03 08:22:52 Using internal plugin for docker-save
2017/08/03 08:22:52 Using internal plugin for docker-tag
2017/08/03 08:22:52 Using internal plugin for shell-local
2017/08/03 08:22:52 Using internal plugin for compress
2017/08/03 08:22:52 Using internal plugin for docker-import
2017/08/03 08:22:52 Using internal plugin for googlecompute-export
2017/08/03 08:22:52 Using internal plugin for vagrant
2017/08/03 08:22:52 Using internal plugin for vagrant-cloud
2017/08/03 08:22:52 Using internal plugin for alicloud-import
2017/08/03 08:22:52 Using internal plugin for artifice
2017/08/03 08:22:52 Using internal plugin for atlas
2017/08/03 08:22:52 Using internal plugin for vsphere
2017/08/03 08:22:52 Using internal plugin for checksum
2017/08/03 08:22:52 Using internal plugin for docker-push
2017/08/03 08:22:52 Using internal plugin for manifest
2017/08/03 08:22:52 Detected home directory from env var: /home/userxyz
2017/08/03 08:22:52 Attempting to open config file: /home/userxyz/.packerconfig
2017/08/03 08:22:52 [WARN] Config file doesn't exist: /home/userxyz/.packerconfig
2017/08/03 08:22:52 Packer config: &{DisableCheckpoint:false DisableCheckpointSignature:false PluginMinPort:10000 PluginMaxPort:25000 Builders:map[amazon-ebsvolume:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-amazon-ebsvolume amazon-instance:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-amazon-instance profitbricks:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-profitbricks null:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-null oneandone:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-oneandone openstack:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-openstack amazon-ebs:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-amazon-ebs file:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-file alicloud-ecs:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-alicloud-ecs amazon-ebssurrogate:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-amazon-ebssurrogate docker:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-docker googlecompute:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-googlecompute hyperv-iso:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-hyperv-iso cloudstack:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-cloudstack qemu:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-qemu triton:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-triton virtualbox-iso:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-virtualbox-iso digitalocean:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-digitalocean parallels-iso:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-parallels-iso virtualbox-ovf:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-virtualbox-ovf vmware-vmx:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-vmware-vmx azure-arm:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-azure-arm vmware-iso:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-vmware-iso amazon-chroot:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-amazon-chroot parallels-pvm:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-builder-parallels-pvm] PostProcessors:map[docker-tag:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-docker-tag vagrant-cloud:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-vagrant-cloud checksum:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-checksum docker-push:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-docker-push manifest:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-manifest docker-save:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-docker-save docker-import:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-docker-import artifice:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-artifice atlas:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-atlas amazon-import:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-amazon-import googlecompute-export:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-googlecompute-export vagrant:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-vagrant alicloud-import:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-alicloud-import shell-local:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-shell-local compress:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-compress vsphere:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-post-processor-vsphere] Provisioners:map[ansible:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-provisioner-ansible puppet-masterless:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-provisioner-puppet-masterless powershell:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-provisioner-powershell shell:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-provisioner-shell converge:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-provisioner-converge file:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-provisioner-file ansible-local:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-provisioner-ansible-local chef-client:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-provisioner-chef-client windows-restart:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-provisioner-windows-restart shell-local:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-provisioner-shell-local windows-shell:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-provisioner-windows-shell puppet-server:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-provisioner-puppet-server salt-masterless:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-provisioner-salt-masterless chef-solo:/usr/local/bin/packer-PACKERSPACE-plugin-PACKERSPACE-packer-provisioner-chef-solo]}
2017/08/03 08:22:52 Detected home directory from env var: /home/userxyz
2017/08/03 08:22:52 Setting cache directory: /home/userxyz/amazon-ebs-elasticbeanstalk/packer_cache
2017/08/03 08:22:52 Loading builder: amazon-ebs
2017/08/03 08:22:52 Plugin could not be found. Checking same directory as executable.
2017/08/03 08:22:52 Current exe path: /usr/local/bin/packer
2017/08/03 08:22:52 Creating plugin client for path: /usr/local/bin/packer
2017/08/03 08:22:52 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-builder-amazon-ebs"}
2017/08/03 08:22:52 Detected home directory from env var: /home/userxyz
2017/08/03 08:22:52 Waiting for RPC address for: /usr/local/bin/packer
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 [INFO] Packer version: 1.0.3
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 Packer Target OS/Arch: linux amd64
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 Built with Go Version: go1.8.1
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 Detected home directory from env var: /home/userxyz
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 Attempting to open config file: /home/userxyz/.packerconfig
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 [WARN] Config file doesn't exist: /home/userxyz/.packerconfig
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 Packer config: &{DisableCheckpoint:false DisableCheckpointSignature:false PluginMinPort:10000 PluginMaxPort:25000 Builders:map[] PostProcessors:map[] Provisioners:map[]}
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 Detected home directory from env var: /home/userxyz
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 Setting cache directory: /home/userxyz/amazon-ebs-elasticbeanstalk/packer_cache
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 args: []string{"packer-builder-amazon-ebs"}
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 Plugin minimum port: 10000
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 Plugin maximum port: 25000
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 Plugin address: unix /tmp/packer-plugin934513007
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 Waiting for connection...
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 Detected home directory from env var: /home/userxyz
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 Serving a plugin connection...
2017/08/03 08:22:52 ui: amazon-ebs output will be in this color.
2017/08/03 08:22:52 ui: 
2017/08/03 08:22:52 Build debug mode: false
2017/08/03 08:22:52 Force build: false
2017/08/03 08:22:52 On error: 
2017/08/03 08:22:52 Preparing build: amazon-ebs
amazon-ebs output will be in this color.
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 Config: {PackerConfig:{PackerBuildName:amazon-ebs PackerBuilderType:amazon-ebs PackerDebug:false PackerForce:false PackerOnError: PackerUserVars:map[region:us-east-1 ami_name_prefix:myinc-ElasticBeanstalk-CARS-Encr source_ami_prefix:aws-elasticbeanstalk-amzn-*.x86_64-tomcat8java8-hvm-* kms_key_id:f841f25b-8694-48a7-be77-8a756df5dbf2]} AccessConfig:{AccessKey: CustomEndpointEc2: MFACode: ProfileName: RawRegion:us-east-1 SecretKey: SkipValidation:false Token: session:<nil>} AMIConfig:{AMIName:myinc-ElasticBeanstalk-CARS-Encr 2017-08-03 12-22-52 AMIDescription:Created by Packer using amazon-ebs builder AMIVirtType: AMIUsers:[] AMIGroups:[] AMIProductCodes:[] AMIRegions:[] AMISkipRegionValidation:false AMITags:map[] AMIEnhancedNetworking:false AMIForceDeregister:false AMIForceDeleteSnapshot:false AMIEncryptBootVolume:true AMIKmsKeyId:f841f25b-8694-48a7-be77-8a756df5dbf2 AMIRegionKMSKeyIDs:map[] SnapshotTags:map[] SnapshotUsers:[] SnapshotGroups:[]} BlockDevices:{AMIBlockDevices:{AMIMappings:[]} LaunchBlockDevices:{LaunchMappings:[]}} RunConfig:{AssociatePublicIpAddress:false AvailabilityZone: EbsOptimized:false IamInstanceProfile: InstanceType:m3.medium RunTags:map[] SourceAmi: SourceAmiFilter:{Filters:map[0xc4202c02a0:0xc4202c02c0 0xc4202c02e0:0xc4202c0300 0xc4202c0320:0xc4202c0340] Owners:[0xc4202c03b0] MostRecent:true} SpotPrice: SpotPriceAutoProduct: DisableStopInstance:false SecurityGroupId: SecurityGroupIds:[] SubnetId: TemporaryKeyPairName:packer_5983159c-a49d-8781-1a4c-572775d726be UserData: UserDataFile:elasticbeanstalk_amzn_linux.userdata WindowsPasswordTimeout:20m0s VpcId: InstanceInitiatedShutdownBehavior:stop Comm:{Type:none SSHHost: SSHPort:0 SSHUsername: SSHPassword: SSHPrivateKey: SSHPty:false SSHTimeout:0s SSHAgentAuth:false SSHDisableAgent:false SSHHandshakeAttempts:0 SSHBastionHost: SSHBastionPort:0 SSHBastionAgentAuth:false SSHBastionUsername: SSHBastionPassword: SSHBastionPrivateKey: SSHFileTransferMethod: WinRMUser: WinRMPassword: WinRMHost: WinRMPort:0 WinRMTimeout:0s WinRMUseSSL:false WinRMInsecure:false WinRMUseNTLM:false WinRMTransportDecorator:<nil>} SSHKeyPairName: SSHPrivateIp:false} VolumeRunTags:map[] ctx:{Data:<nil> Funcs:map[clean_ami_name:0xc1c080] UserVariables:map[region:us-east-1 ami_name_prefix:myinc-ElasticBeanstalk-CARS-Encr source_ami_prefix:aws-elasticbeanstalk-amzn-*.x86_64-tomcat8java8-hvm-* kms_key_id:f841f25b-8694-48a7-be77-8a756df5dbf2] EnableEnv:false BuildName:amazon-ebs BuildType:amazon-ebs TemplatePath:/home/userxyz/amazon-ebs-elasticbeanstalk/elasticbeanstalk.json}}
2017/08/03 08:22:52 packer: 2017/08/03 08:22:52 [ERR] Checkpoint error: Get https://checkpoint-api.hashicorp.com/v1/check/packer?arch=amd64&os=linux&signature=f5bf8867-c2f0-b493-7004-0375c26d931f&version=1.0.3: Forbidden
2017/08/03 08:22:52 [ERR] Checkpoint error: Get https://checkpoint-api.hashicorp.com/v1/check/packer?arch=amd64&os=linux&signature=f5bf8867-c2f0-b493-7004-0375c26d931f&version=1.0.3: Forbidden
2017/08/03 08:22:52 Waiting on builds to complete...
2017/08/03 08:22:52 Starting build run: amazon-ebs
2017/08/03 08:22:52 Running builder: amazon-ebs
2017/08/03 08:22:52 [INFO] (telemetry) Starting builder amazon-ebs

2017/08/03 08:22:52 ui: ==> amazon-ebs: Prevalidating AMI Name...
==> amazon-ebs: Prevalidating AMI Name...
2017/08/03 08:22:56 packer: 2017/08/03 08:22:56 Using AMI Filters {
2017/08/03 08:22:56 packer:   Filters: [{
2017/08/03 08:22:56 packer:       Name: "root-device-type",
2017/08/03 08:22:56 packer:       Values: ["ebs"]
2017/08/03 08:22:56 packer:     },{
2017/08/03 08:22:56 packer:       Name: "virtualization-type",
2017/08/03 08:22:56 packer:       Values: ["hvm"]
2017/08/03 08:22:56 packer:     },{
2017/08/03 08:22:56 packer:       Name: "name",
2017/08/03 08:22:56 packer:       Values: ["aws-elasticbeanstalk-amzn-*.x86_64-tomcat8java8-hvm-*"]
2017/08/03 08:22:56 packer:     }],
2017/08/03 08:22:56 packer:   Owners: ["amazon"]
2017/08/03 08:22:56 packer: }
2017/08/03 08:23:19 ui:     amazon-ebs: Found Image ID: ami-7a171f6c
    amazon-ebs: Found Image ID: ami-7a171f6c
2017/08/03 08:23:19 ui: ==> amazon-ebs: Creating temporary keypair: packer_5983159c-a49d-8781-1a4c-572775d726be
==> amazon-ebs: Creating temporary keypair: packer_5983159c-a49d-8781-1a4c-572775d726be
2017/08/03 08:23:19 ui: ==> amazon-ebs: Deleting temporary keypair...
==> amazon-ebs: Deleting temporary keypair...
2017/08/03 08:23:19 packer: Build 'amazon-ebs' errored: unexpected EOF

==> Some builds didn't complete successfully and had errors:
--> amazon-ebs: unexpected EOF

==> Builds finished but no artifacts were created.
panic: port must be set to a non-zero value.
2017/08/03 08:23:19 packer: 
2017/08/03 08:23:19 packer: goroutine 83 [running]:
2017/08/03 08:23:19 packer: github.com/hashicorp/packer/builder/amazon/common.(*StepSecurityGroup).Run(0xc420279d00, 0x23357a0, 0xc4202b1d40, 0x0)
2017/08/03 08:23:19 packer:     /Users/mmarsh/go/src/github.com/hashicorp/packer/builder/amazon/common/step_security_group.go:48 +0x10db
2017/08/03 08:23:19 packer: github.com/hashicorp/packer/vendor/github.com/mitchellh/multistep.(*BasicRunner).Run(0xc420279d80, 0x23357a0, 0xc4202b1d40)
2017/08/03 08:23:19 packer:     /Users/mmarsh/go/src/github.com/hashicorp/packer/vendor/github.com/mitchellh/multistep/basic_runner.go:70 +0x1e1
2017/08/03 08:23:19 packer: github.com/hashicorp/packer/builder/amazon/ebs.(*Builder).Run(0xc420218500, 0x2338480, 0xc4202cad40, 0x2330420, 0xc4202c1d00, 0x2336ee0, 0xc42000c488, 0xc4202b1a40, 0x0, 0xc4202424e0, ...)
2017/08/03 08:23:19 packer:     /Users/mmarsh/go/src/github.com/hashicorp/packer/builder/amazon/ebs/builder.go:220 +0x15b9
2017/08/03 08:23:19 packer: github.com/hashicorp/packer/packer/rpc.(*BuilderServer).Run(0xc420209860, 0x1, 0xc4202c1c2c, 0x0, 0x0)
2017/08/03 08:23:19 packer:     /Users/mmarsh/go/src/github.com/hashicorp/packer/packer/rpc/builder.go:93 +0x1ea
2017/08/03 08:23:19 packer: reflect.Value.call(0xc4202424e0, 0xc42000c360, 0x13, 0x193b13f, 0x4, 0xc420041f20, 0x3, 0x3, 0x0, 0x0, ...)
2017/08/03 08:23:19 packer:     /usr/local/go/src/reflect/value.go:434 +0x91f
2017/08/03 08:23:19 packer: reflect.Value.Call(0xc4202424e0, 0xc42000c360, 0x13, 0xc4202d4720, 0x3, 0x3, 0x0, 0x0, 0x0)
2017/08/03 08:23:19 packer:     /usr/local/go/src/reflect/value.go:302 +0xa4
2017/08/03 08:23:19 packer: net/rpc.(*service).call(0xc420212c80, 0xc420212c40, 0xc420207980, 0xc420232600, 0xc420209ca0, 0x15a9620, 0xc4202c1c28, 0x18a, 0x1574120, 0xc4202c1c2c, ...)
2017/08/03 08:23:19 packer:     /usr/local/go/src/net/rpc/server.go:387 +0x144
2017/08/03 08:23:19 packer: created by net/rpc.(*Server).ServeCodec
2017/08/03 08:23:19 packer:     /usr/local/go/src/net/rpc/server.go:481 +0x404
2017/08/03 08:23:19 [INFO] (telemetry) ending amazon-ebs
2017/08/03 08:23:19 [INFO] (telemetry) found error: unexpected EOF
2017/08/03 08:23:19 ui error: Build 'amazon-ebs' errored: unexpected EOF
2017/08/03 08:23:19 Builds completed. Waiting on interrupt barrier...
2017/08/03 08:23:19 machine readable: error-count []string{"1"}
2017/08/03 08:23:19 ui error: 
==> Some builds didn't complete successfully and had errors:
2017/08/03 08:23:19 machine readable: amazon-ebs,error []string{"unexpected EOF"}
2017/08/03 08:23:19 ui error: --> amazon-ebs: unexpected EOF
2017/08/03 08:23:19 ui: 
==> Builds finished but no artifacts were created.
2017/08/03 08:23:19 /usr/local/bin/packer: plugin process exited
2017/08/03 08:23:19 [WARN] (telemetry) Error finalizing report. This is safe to ignore. Post https://checkpoint-api.hashicorp.com/v1/telemetry/packer: Forbidden
2017/08/03 08:23:19 waiting for all plugin processes to complete...
panic: port must be set to a non-zero value.
2017/08/03 08:23:19 packer: 
2017/08/03 08:23:19 packer: goroutine 83 [running]:
2017/08/03 08:23:19 packer: github.com/hashicorp/packer/builder/amazon/common.(*StepSecurityGroup).Run(0xc420279d00, 0x23357a0, 0xc4202b1d40, 0x0)
2017/08/03 08:23:19 packer:     /Users/mmarsh/go/src/github.com/hashicorp/packer/builder/amazon/common/step_security_group.go:48 +0x10db
2017/08/03 08:23:19 packer: github.com/hashicorp/packer/vendor/github.com/mitchellh/multistep.(*BasicRunner).Run(0xc420279d80, 0x23357a0, 0xc4202b1d40)
2017/08/03 08:23:19 packer:     /Users/mmarsh/go/src/github.com/hashicorp/packer/vendor/github.com/mitchellh/multistep/basic_runner.go:70 +0x1e1
2017/08/03 08:23:19 packer: github.com/hashicorp/packer/builder/amazon/ebs.(*Builder).Run(0xc420218500, 0x2338480, 0xc4202cad40, 0x2330420, 0xc4202c1d00, 0x2336ee0, 0xc42000c488, 0xc4202b1a40, 0x0, 0xc4202424e0, ...)
2017/08/03 08:23:19 packer:     /Users/mmarsh/go/src/github.com/hashicorp/packer/builder/amazon/ebs/builder.go:220 +0x15b9
2017/08/03 08:23:19 packer: github.com/hashicorp/packer/packer/rpc.(*BuilderServer).Run(0xc420209860, 0x1, 0xc4202c1c2c, 0x0, 0x0)
2017/08/03 08:23:19 packer:     /Users/mmarsh/go/src/github.com/hashicorp/packer/packer/rpc/builder.go:93 +0x1ea
2017/08/03 08:23:19 packer: reflect.Value.call(0xc4202424e0, 0xc42000c360, 0x13, 0x193b13f, 0x4, 0xc420041f20, 0x3, 0x3, 0x0, 0x0, ...)
2017/08/03 08:23:19 packer:     /usr/local/go/src/reflect/value.go:434 +0x91f
2017/08/03 08:23:19 packer: reflect.Value.Call(0xc4202424e0, 0xc42000c360, 0x13, 0xc4202d4720, 0x3, 0x3, 0x0, 0x0, 0x0)
2017/08/03 08:23:19 packer:     /usr/local/go/src/reflect/value.go:302 +0xa4
2017/08/03 08:23:19 packer: net/rpc.(*service).call(0xc420212c80, 0xc420212c40, 0xc420207980, 0xc420232600, 0xc420209ca0, 0x15a9620, 0xc4202c1c28, 0x18a, 0x1574120, 0xc4202c1c2c, ...)
2017/08/03 08:23:19 packer:     /usr/local/go/src/net/rpc/server.go:387 +0x144
2017/08/03 08:23:19 packer: created by net/rpc.(*Server).ServeCodec
2017/08/03 08:23:19 packer:     /usr/local/go/src/net/rpc/server.go:481 +0x404
2017/08/03 08:23:19 [INFO] (telemetry) ending amazon-ebs
2017/08/03 08:23:19 [INFO] (telemetry) found error: unexpected EOF
2017/08/03 08:23:19 ui error: Build 'amazon-ebs' errored: unexpected EOF
2017/08/03 08:23:19 Builds completed. Waiting on interrupt barrier...
2017/08/03 08:23:19 machine readable: error-count []string{"1"}
2017/08/03 08:23:19 ui error: 
==> Some builds didn't complete successfully and had errors:
2017/08/03 08:23:19 machine readable: amazon-ebs,error []string{"unexpected EOF"}
2017/08/03 08:23:19 ui error: --> amazon-ebs: unexpected EOF
2017/08/03 08:23:19 ui: 
==> Builds finished but no artifacts were created.
2017/08/03 08:23:19 /usr/local/bin/packer: plugin process exited
2017/08/03 08:23:19 [WARN] (telemetry) Error finalizing report. This is safe to ignore. Post https://checkpoint-api.hashicorp.com/v1/telemetry/packer: Forbidden
2017/08/03 08:23:19 waiting for all plugin processes to complete...

Failed to report panic. This is safe to ignore: Post https://checkpoint-api.hashicorp.com/v1/telemetry/packer: Forbidden

!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Packer crashed! This is always indicative of a bug within Packer.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Packer[1] so that we can fix this.

[1]: https://github.com/hashicorp/packer/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

--- a/builder/amazon/common/step_security_group.go
+++ b/builder/amazon/common/step_security_group.go
@@ -45,7 +45,9 @@ func (s *StepSecurityGroup) Run(state multistep.StateBag) multistep.StepAction {
 
 	port := s.CommConfig.Port()
 	if port == 0 {
-		panic("port must be set to a non-zero value.")
+		if s.CommConfig.Type != "none" {
+			panic("port must be set to a non-zero value.")
+		}
 	}
 
 	// Create the group

